src/util/signature.rs

```diff
@@ -0,0 +1,34 @@
+use base64::{engine::general_purpose::STANDARD as base64, Engine};
+use hmac::{Hmac, Mac};
+use sha2::Sha256;
+use url::form_urlencoded;
+
+// Generate signature with timestamp as required by Cloudflare's is_timed_hmac_valid_v0 function
+// https://developers.cloudflare.com/ruleset-engine/rules-language/functions/#hmac-validation
+pub fn timed_hmac(message: &str, secret: &str, timestamp: usize) -> String {
```

Comment:
It's not used by the worker, is it? If the only reason for this file is to be exactly the same implementation as in the scheduler, let's leave a TODO for moving it into a common crate

Reply:
done